src/nouveau/src/nouveau_fabric_search.erl

```diff
@@ -215,11 +222,43 @@ convert_item(Item) ->
             maps:get(<<"value">>, Item)
     end.
 
-merge_facets(FacetsA, null, _Limit) ->
+top_n_facets(SearchResults, TopN) ->
+    After = SearchResults#{
+        <<"counts">> => top_n_facets(<<"counts">>, SearchResults, TopN),
```

Comment:
If we're updating existing keys, it's usually better to assert that with `:=`.

Reply:
They might not exist.

Reply:
Ah you're right, I missed that. In that case disregard my silliness.